src/walk-manifest.js

```diff
@@ -147,125 +168,160 @@ const walkPlaylist = function(options) {
       visitedUrls = [],
       requestTimeout = 1500,
       requestRetryMaxAttempts = 5,
+      dashPlaylist = null,
       requestRetryDelay = 5000
     } = options;
 
     let resources = [];
-    const manifest = {};
+    const manifest = {parent};
 
     manifest.uri = uri;
     manifest.file = path.join(basedir, fsSanitize(path.basename(uri)));
 
     // if we are not the master playlist
     if (parent) {
-      manifest.file = path.join(
-        path.dirname(parent.file),
-        'manifest' + manifestIndex,
-        fsSanitize(path.basename(manifest.file))
-      );
-      // get the real uri of this playlist
-      if (!isAbsolute(manifest.uri)) {
-        manifest.uri = joinURI(path.dirname(parent.uri), manifest.uri);
+      if (!dashPlaylist) {
+        manifest.file = path.join(
+          path.dirname(parent.file),
+          'manifest' + manifestIndex,
+          fsSanitize(path.basename(manifest.file))
+        );
+        // get the real uri of this playlist
+        if (!isAbsolute(manifest.uri)) {
+          manifest.uri = joinURI(path.dirname(parent.uri), manifest.uri);
+        }
+        // replace original uri in file with new file path
+        parent.content = Buffer.from(parent.content.toString().replace(uri, path.relative(path.dirname(parent.file), manifest.file)));
+      } else if (dashPlaylist.segments && dashPlaylist.segments.length) {
+        manifest.file = path.join(path.dirname(getMaster(parent).file), fsSanitize(path.dirname(dashPlaylist.segments[0].uri)));
```

Comment:
Somewhat hacky, and mpd-parser should probably return a property for the folder name of it's "playlists" but using the dirname of the first segment works for now.